### PR TITLE
Makes the dev-scripts and readme easier to understand / use

### PR DESCRIPTION
**Pull request by davidfestal**
Main changes:
- allow using individual scripts more easily for deployment of already existing images, etc ... This allows covering use-cases that were not covered until now.
- rename some scripts to make them more meaningful, and make clear what is related to the minishift docker environment.
- Change the Readme to move the Maven section at the end and make the script section more structured.

Signed-off-by: David Festal <dfestal@redhat.com>

### CODE REVIEW

File: dev-scripts/local_docker_create_images_and_tag.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 export DeveloperBuild="true"
 
-currentDir = `pwd`
+currentDir=`pwd`
```
Comment (benoitf) [inline dev-scripts/local_docker_create_images_and_tag.sh]: shouldn't we use newer form of calling with $, `$(pwd)` call ?

Comment (benoitf) [inline dev-scripts/local_docker_create_images_and_tag.sh]: BTW you may use https://www.shellcheck.net/ to get other review on shell scripts.

**Comment by davidfestal** [comment]
@mlabuda does it now cover the use-cases that you found were not covered previously ?

File: README.md [hunk 2]
```diff
@@ -3,8 +3,8 @@
 ## Table Of Content
 
 * [What is the RedHat Che distribution](#what-is-the-redhat-che-distribution)
-* [RedHat Che distribution Maven build](#redhat-che-distribution-maven-build)
 * [Developer-friendly scripts to build and deploy or rollout to minishift](#developer-friendly-scripts-to-build-and-deploy-or-rollout-to-minishift)
+* [Maven build details](#maven-build-details)
```
Comment (l0rd) [inline README.md]: I think that we should have the following sections here:
1. What is the Red Hat Che distribution
2. How to build it
3. How to deploy it
4. Further details

"Developer-friendly scripts to build and deploy or rollout to minishift" is too long and should be splitted in how to build it and how to deploy it.

File: README.md [hunk 3]
```diff
@@ -3,8 +3,8 @@
...
 
 ## TL;DR
```
Comment (l0rd) [inline README.md]: This TL;DR doesn't make sense as it is because it just covers the build. You can:
- remove it
- make it a 2 snippets: to build, to deploy. The goal is that people in a hurry can just copy/paste without going through the doc.

I would remove it.

File: README.md [hunk 4]
```diff
@@ -42,8 +42,242 @@ RedHat modifications against the upstream Che include:
 - Keycloak integration
 - [fabric8-analytics Language Server](https://github.com/fabric8-analytics/fabric8-analytics-lsp-server) 
 
+## Developer-friendly scripts to build and deploy or rollout to minishift
+
+The `dev-scripts` folder contains various scripts for developers that allow easily running the Maven
```
Comment (l0rd) [inline README.md]: I find it still complicated. What you should do is.

### How to build it

```bash
## build pre-requisites

## set some env vars

## build
./dev-scripts/build_fabric8.sh
```

`build_fabric8.sh` options:
- `-Dfast`: description...
- 
- 
....

File: dev-scripts/local_docker_create_images_and_tag.sh (hunk 1 above)
Comment (l0rd) [inline dev-scripts/local_docker_create_images_and_tag.sh]: 👍  @benoitf

File: dev-scripts/build_fabric8.sh [hunk 5]
```diff
@@ -16,25 +16,35 @@ else
...
     echo "! Using dedicated Upstream Che repo CHECKED-OUT in the following target directory : "
     echo "!     $(pwd)/target/export/che-dependencies/che "
     echo "!"
-    echo "! Docker image to deploy: ${CHE_IMAGE_REPO}:${CHE_IMAGE_TAG}"
+    echo "! Building ${keycloakSupport}"
     echo "!"
     echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
-    additionalArgument="-DwithoutKeycloak"
+    echo "!"
+        additionalArgument="-DwithoutKeycloak"
```
Comment (l0rd) [inline dev-scripts/build_fabric8.sh]: formatting?

**Comment by davidfestal** [comment]
@l0rd I made changes according to your comments
 
> after the PR the README.md would be longer than before.

This is not the case anymore. it's (slightly) smaller than before with more use-cases covered and further details pushed to the end.

File: README.md [hunk 6]
```diff
@@ -42,8 +19,230 @@ RedHat modifications against the upstream Che include:
...
+
+    The docker namespace used by default is `docker.io/rhchestage`
+    
+### Build RedHat distribution
+
+If you want to build and deploy the RH distribution to Minishift in one go, you
+might be also be interested by the [*all-in-one scripts*](#all-in-one-scripts-for-minishift)
+
+```bash
+dev-scripts/build_fabric8.sh
```
Comment (l0rd) [inline README.md]: Why not `dev-scripts/build_fabric8.sh [PARAMETERS]` ?

File: README.md [hunk 7]
```diff
@@ -42,8 +19,230 @@ RedHat modifications against the upstream Che include:
...
+    ```bash
+    export DOCKER_HUB_NAMESPACE=myDockerNamspace
+    ```
+
+    The docker namespace used by default is `docker.io/rhchestage`
+    
+### Build RedHat distribution
+
+If you want to build and deploy the RH distribution to Minishift in one go, you
+might be also be interested by the [*all-in-one scripts*](#all-in-one-scripts-for-minishift)
```
Comment (l0rd) [inline README.md]: I think you can remove this line.

File: README.md [hunk 8]
```diff
@@ -42,8 +19,230 @@ RedHat modifications against the upstream Che include:
...
+This can be done with the following command:
+
+```bash
+minishift start
+```
+
+#### Deploy to Minishift 
+
+__Important__: Before deploying to Minishift, ensure that you created / tagged the Che docker images 
+*in the Minishift docker environment* (see [previous section](#in-the-minishift-docker-environment))
```
Comment (l0rd) [inline README.md]: I'm not sure that it's completely true. Users that just want to deploy to minishift (without building Che first) don't need to create/tag Che Docker images. I think that will be the majority of people that are going to read this section. So I would move this sentence as a warning a few lines below.

File: cico_do_docker_build_tag_push.sh [hunk 9]
```diff
@@ -12,12 +12,14 @@ upstreamCheRepoFullPath=`pwd`
...
 
 # Now lets build the local docker images
-mkdir ${currentDir}/target/docker
+mkdir ${currentDir}/target/docker 2>/dev/null
 cp -R dockerfiles ${currentDir}/target/docker
 
 cd ${currentDir}/target/docker/dockerfiles/che
 cat Dockerfile.centos > Dockerfile
 
+createTags=""
```
Comment (l0rd) [inline cico_do_docker_build_tag_push.sh]: This variable doesn't seems to be used somewhere else

Comment (davidfestal) [inline cico_do_docker_build_tag_push.sh]: it's used in line [88](https://github.com/redhat-developer/rh-che/pull/112/files#diff-6aee2f002a3f356e002c5ef80242643cR88), to show, at the end of the command, a summary of what docker images were tagged and in which environment.

File: README.md (hunk 6 above)
Comment (davidfestal) [inline README.md]: yes, let me change this

**Comment by davidfestal** [comment]
And now, is it OK @l0rd ?

File: cico_do_docker_build_tag_push.sh (hunk 9 above)
Comment (l0rd) [inline cico_do_docker_build_tag_push.sh]: @davidfestal  at line 88 I can see `dockerTags`. Bu I was talking about `crateTags` variable.

**Comment by davidfestal** [comment]
@FYI I just tested the `cico_build.sh` on a raw CentOS VM and the build is successful.
